test/yargs.js

```diff
@@ -252,4 +264,47 @@ describe('yargs dsl tests', function () {
       yargs.terminalWidth().should.be.gt(0)
     })
   })
+
+  describe('choices', function () {
+    it('accepts an object', function () {
+      var optChoices = yargs([])
+        .choices({
+          color: ['red', 'green', 'blue'],
+          stars: [1, 2, 3, 4, 5]
+        })
+        .choices({
+          size: ['xl', 'l', 'm', 's', 'xs']
+        })
+        .getOptions().choices
+
+      optChoices.should.deep.equal({
+        color: ['red', 'green', 'blue'],
+        stars: [1, 2, 3, 4, 5],
+        size: ['xl', 'l', 'm', 's', 'xs']
+      })
+    })
+
+    it('accepts a string and array', function () {
+      var optChoices = yargs([])
+        .choices('meat', ['beef', 'chicken', 'pork', 'bison'])
+        .choices('temp', ['rare', 'med-rare', 'med', 'med-well', 'well'])
+        .getOptions().choices
```

Comment:
I like the use of the `getOptions()` method for building out these assertions :100: